src/web/rustdoc.rs

```diff
@@ -215,6 +183,22 @@ pub fn rustdoc_redirector_handler(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Serialize)]
+struct RustdocPage {
+    latest_path: String,
+    latest_version: String,
+    inner_path: String,
+    is_latest_version: bool,
+    rustdoc_head: String,
+    rustdoc_body: String,
+    rustdoc_body_class: String,
+    krate: CrateDetails,
+}
+
+impl_webpage! {
+    RustdocPage = "rustdoc/page.html",
+}
+
```

Comment:
Can you move this above `RustLangRedirector` where it was before? It would make the diff smaller (and also there's no reason to change it).

Reply:
I put it here because it's right above `rustdoc_html_server_handler`, which is where it's used. I did this throughout the PR because I think it makes better sense and has better browse-locality to put the struct next to where it's used